[Feat] 대학교 이메일 입력 도메인 & 네트워킹

tea-hkim opened the pull request:
## 작업내용
<!-- 작업 내용과 이미지를 첨부해주세요. -->
- [x] DomainSettingUseCase 생성 
- [x] DomainSettingUseCase 연결
<!-- (+스크린샷)이 있다면 적어주세요. 없으면 지워주세요-->
|작업 전|작업 후|
|:---:|:---:|
|<img width="250" src="">|<img width="250" src="https://user-images.githubusercontent.com/81206228/200221338-1b74c7de-4361-4322-b546-47d6318d4d42.gif">|

### DomainSettingUseCase 연결
 - 기존 UserSignupUseCase에 있던 postSingupUser를 삭제하였습니다.
 - DomainSettingUseCase를 만들어 postSingupUser를 postUserEmail로 변경하였습니다
 - API문서 그대로 postUserEmail의 경우 user api 영역에 있으므로 UserAPI에 남겨뒀습니다.
   하지만 프론트엔드에서는 domainSetting의 영역이기 때문에 따로 DomainSettingUseCase를 만들어 UserAPI에서 호출하였습니다

###  DomainSettingUseCase 생성 
 - DomainSettingViewController와 DomainSettingUseCase를 연결했습니다
 - DomainSettingViewController와 OrganizationListViewController를 연결했습니다
   pushViewController를 통해 넘어오기 전, DomainSettingViewController의 파라미터 값으로 가지고 있던 Organization을 넘겨주도록 하였습니다

## 리뷰포인트
<!-- 리뷰가 필요한 포인트와 해당 되는 커밋을 링크로 걸어주세요. -->

## Checklist
- [ ] 브랜치를 가져와 작업한 경우 이전 브랜치에 PR을 보냈는지 확인
- [x] 빌드를 위해 SceneDelegate 수정한 것 PR로 올리지 않았는지 확인
- [x] 필요없는 주석, 프린트문 제거했는지 확인
- [x] 컨벤션 지켰는지 확인
- [x] final, private 제대로 넣었는지 확인
- [x] 다양한 디바이스에 레이아웃이 대응되는지 확인
  - [x] iPhone SE
  - [x] iPhone 13
  - [x] iPhone 13 Pro Max

tea-hkim committed:
Feat:sparkles:: DomainSettingUseCase 생성

 - 기존 UserSignupUseCase에 있던 postSingupUser를 삭제하였습니다.
 - DomainSettingUseCase를 만들어 postSingupUser를 postUserEmail로 변경하였습니다
 - API문서 그대로 postUserEmail의 경우 user api 영역에 있으므로 UserAPI에 남겨뒀습니다.
   하지만 프론트엔드에서는 domainSetting의 영역이기 때문에 따로 DomainSettingUseCase를 만들어 UserAPI에서 호출하였습니다

tea-hkim committed:
Feat:sparkles:: DomainSettingUseCase 연결

 - DomainSettingViewController와 DomainSettingUseCase를 연결했습니다
 - DomainSettingViewController와 OrganizationListViewController를 연결했습니다
   pushViewController를 통해 넘어오기 전, DomainSettingViewController의 파라미터 값으로 가지고 있던 Organization을 넘겨주도록 하였습니다

Gobans commented on Projects/App/Sources/Domain/Organization/DomainSettingUseCase.swift:
```diff
@@ -0,0 +1,42 @@
+                case .finished: break
+                }
+            } receiveValue: { [weak self] _ in
+                print()
```
p1
잘못 들어간듯요!

tea-hkim replied:
변경했습니다! f3c5ec458a8efe52b37b4209ee53115efa03c9b9

Gobans commented on Projects/App/Sources/Domain/Organization/DomainSettingUseCase.swift:
```diff
@@ -0,0 +1,42 @@
+    private var cancelBag = Set<AnyCancellable>()
+    
+    func postUserEmail(email: String, orgnization: Organization) {
+//        guard let authorization = UserDefaults.standard.authToken else { return }
+        let authorization = "eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJ6ZXN0eSIsImlhdCI6MTY2NzE4MDc5MiwiZXhwIjoxNjY4MzkwMzkyLCJzdWIiOiIzNyJ9.jZ35GteW1sM9oz_QQ0cjdxk4p0CSf81Gzv7Kjs_W-zA"
```
요거 로그인 기능 merge 되면 키체인으로 불러와주세요!

tea-hkim replied:
todo로 남겨뒀습니다!
추후에 수정할 예정입니다
938aac912e20b72060c795adc7c097c776e09c61

Gobans commented on Projects/App/Sources/UI/Organization/DomainSetting/View/DomainSettingViewController.swift:
```diff
@@ -106,10 +135,12 @@ extension DomainSettingViewController {
         
         domainPlaceholder.textColor = .white
-        domainPlaceholder.text = "@pos.idserve.net"
+        domainPlaceholder.text = "@\(viewModel.getOrgDomain())"
```
p3
그냥 간단하게 값에 접근하는 것이라면 private를 풀고 바로 접근해도 될 것 같습니다

tea-hkim replied:
변경했습니다! 26e6ee695f697a9321da61504815b159a3b79294

Gobans commented on Projects/App/Sources/UI/Organization/DomainSetting/ViewModel/DomainSettingViewModel.swift:
```diff
@@ -10,32 +10,67 @@ import Combine
-    @Published var isButtonValid = false
-    
-    var orgDomain: String = ""
+    @Published var isInputValid = true
```
p2
isInputValid 가 반대로 사용되고 있는 것 같아요.
valid 하다면 true, 아니라면 false 로 설정 해주세요

tea-hkim replied:
변경했습니다! 2eb9a777641abbae9462e17352dbaf3cd8e39115

Gobans reviewed (commented):
고생하셨습니다 닉!
리뷰 몇개 남겼는데 확인해주세요
++
시뮬레이터 상에서
<img width="250" alt="image" src="https://user-images.githubusercontent.com/56781342/200256434-1d6f6af4-5b8d-48a9-a43a-b2b090573fce.png">
골뱅이가 두개인데
이건 목데이터라 그런거겠죠?

tea-hkim committed:
Chore:broom:: print 제거

tea-hkim committed:
Chore:broom:: authentication todo 설정

tea-hkim committed:
Chore:broom:: DomainSettingViewModel 내부 Organization 접근권한 변경

 - private을 삭제헀습니다

tea-hkim committed:
Feat:sparkles:: viewWillAppear 추가

 - VerifingCodeViewController로 뷰가 전환되고 나서 다시 현재의 뷰로 돌아올 경우 키보드가 내려가는 현상이 있었습니다
 - 해당 문제를 viewWillAppear 내부에 뷰에 들어오자마자 키보드가 올라오는 코드를 추가하여 해결했습니다.

tea-hkim committed:
Chore:broom:: ViewModel isInputValid 로직 변경

 - 기존의 직관적이지 않은 논리값에서 변수명에 직관적인 논리값 전환으로 변경하였습니다
